### PR TITLE
FEAT: Delete user

### PR DESCRIPTION
**Pull request by haruelrovix**
Part of https://github.com/mazipan/tanyaaja/issues/3

## Description

- Add middleware to handle Authorization
- Add methods to Archive a Page and Delete Questions by `uid`
- Add endpoint to DELETE user

📝 As stated [here](https://nextjs.org/docs/pages/api-reference/edge#unsupported-apis),

> Native Node.js APIs are not supported

that's why there is a new package added `next-firebase-auth-edge` as a workaround.

### CODE REVIEW

File: src/app/api/private/user/delete/route.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import { headers } from 'next/headers'
+import { NextResponse } from 'next/server'
+
+import { deleteUser } from '@/lib/firebase-auth-edge'
+import { archivePage, deleteQuestionsByUid, getUserByUid } from '@/lib/notion'
+
+export async function DELETE() {
+  // Get the User uid
+  const headersInstance = headers()
+  const uid = headersInstance.get('user-id') as string
```
Comment (mazipan) [inline src/app/api/private/user/delete/route.ts]: Should passing an Auth Token.
Make sure it's not accessible by non-login user.

Comment (mazipan) [reply src/app/api/private/user/delete/route.ts]: If possible, do not accept UID from the parameter, get it from the session.
To make sure the user is only executing on their own account

**Comment by mazipan** [comment]
I need to review #129 first ya.

Still not have a time to properly setting up the Firebase Admin.

Maybe next week 🙏

File: src/app/api/private/user/delete/route.ts (hunk 1 above)
Comment (haruelrovix) [reply src/app/api/private/user/delete/route.ts]: The Authentication is handled by middleware now:

```js
export async function middleware(request: NextRequest) {
  try {
    const headersInstance = headers()
    const token = headersInstance.get('Authorization')
```

https://github.com/mazipan/tanyaaja/pull/134/files#diff-0626369278c89781505d2c04865ecce1302e979271bcfb471e029c19addc13a0

The `user-id` is set by middleware as well:

```js
    // Add new request headers
    // Ref: https://vercel.com/templates/next.js/edge-functions-modify-request-header
    const requestHeaders = new Headers(request.headers)
    requestHeaders.set('user-id', decodedToken.uid)
```

https://github.com/mazipan/tanyaaja/pull/134/files#diff-0626369278c89781505d2c04865ecce1302e979271bcfb471e029c19addc13a0R21

---

This is some of the behavior for the DELETE user endpoint x middleware:

**[Scenario 1]**: Request without token

```sh
curl --location --request DELETE ''{{BASE_URL}}/api/private/user/delete'
```

```json
403 Forbidden

{
    "message": "Can not found the session."
}
```

**[Scenario 2]**: Request with **incorrect** token

```sh
curl --location --request DELETE '{{BASE_URL}}/api/private/user/delete' \
--header 'Authorization: ***'
```

```json
401 Unauthorized

{
    "message": "Invalid token signature."
}
```

**[Scenario 2]**: Request with **valid** token

```
204 No Content
```

---

Reference for Next.js middleware [here](https://nextjs.org/docs/app/building-your-application/routing/middleware).

**Pull request merged**